src/tests/test23-tls-protos-tls1_2.sh

```diff
@@ -1,23 +1,21 @@
 #!/bin/sh
 # Test tls-protos in global scope
+
 . hitch_test.sh
-set +o errexit
 
 # only TLSv1.2
-mk_cfg <<EOF
+cat >hitch.cfg <<EOF
 backend = "[hitch-tls.org]:80"
 frontend = "[*]:$LISTENPORT"
 pem-file = "${CERTSDIR}/default.example.com"
 tls-protos = TLSv1.2
 EOF
 
-hitch $HITCH_ARGS --config=$CONFFILE
-test $? -eq 0 || die "Hitch did not start."
+start_hitch --config=hitch.cfg
 
 # this will fail on platforms that have OpenSSL compiled without SSLv3
-openssl s_client -connect $LISTENADDR:$LISTENPORT -tls1_2
-test $? -eq 0 || die "Connecting using TLS 1.2 failed."
+# XXX: find how to detect the lack of SSLv3
```

Comment:
One naive way would be to check for the `-ssl3` flag missing (note how `-no_ssl3` is still there):
`$ openssl s_client -help 2>&1 | grep -Fq -- -ssl3`

Reply:
We don't want to run `s_client` in a sub-shell (referring to a previous comment) and `-ssl3` wasn't used in the the original test, so I'd need to build `openssl`and find out myself. Otherwise the new check is as comprehensive as the old one. I should probably document how to use the test helpers, I was at least planning to...

But we could do something like this:

```
openssl s_client -help 2>&1 |
grep -q -e -ssl3 ||
skip "no ssl3 support"
```

Same as here:

https://github.com/Dridi/hitch/blob/8d796e591db5c06e228f2568b61894af84fe8488/src/tests/test25-dynamic-backend-address.sh#L10-L12